lib/charms/traefik_k8s/v0/ingress_per_unit.py

```diff
@@ -439,10 +455,58 @@ def is_failed(self, relation: Relation = None) -> bool:
 
         return False
 
-    def get_request(self, relation: Relation):
-        """Get the IngressRequest for the given Relation."""
-        provider_app_data, requirer_unit_data = self._fetch_relation_data(relation)
-        return IngressRequest(self, relation, provider_app_data, requirer_unit_data)
+    def is_unit_ready(self, relation: Relation, unit: Unit) -> bool:
+        """Whether the given unit has shared its side of the data."""
+        assert (
```

Comment:
Is this supposed to still be here? Seems like a strange use of `assert`, or at least a strange place to use it

Reply:
It's a sanity check, that condition should not occur in production (is_unit_ready is used in ways which guard against that ever happening), but I had a case during testing (because of https://github.com/canonical/traefik-k8s-operator/issues/39) where this precondition was violated. I think it makes sense to check it because the error is really hard to track down otherwise.

Will add a comment.

Reply:
> The current code generator emits no code for an assert statement when optimization is requested at compile time. 

https://docs.python.org/3/reference/simple_stmts.html#the-assert-statement

So should be fine.